### PR TITLE
[Merged by Bors] - feat: absolute convergence from conditional of power series

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
From https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Bounds.20on.20alternating.20convergent.20series/near/417504820

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 1]
```diff
@@ -562,6 +570,34 @@ theorem not_summable_of_ratio_test_tendsto_gt_one {α : Type*} [SeminormedAddCom
...
+
+variable [NormedDivisionRing α] [CompleteSpace α] {f : ℕ → α}
+
+/-- If a power series converges at `w`, it converges absolutely at all `z` of lesser norm. -/
+theorem summable_power_of_norm_lt {w z : α}
+    (h : CauchySeq fun n ↦ ∑ i in range n, f i * w ^ i) (hz : ‖z‖ < ‖w‖) :
+    Summable fun n ↦ f n * z ^ n := by
+  -- First show `0 < ‖w‖`
+  cases' (norm_nonneg w).eq_or_gt with hw hw
+  · exact absurd ((norm_nonneg z).trans_lt (hw ▸ hz)) (lt_irrefl 0)
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:583]:
```suggestion
  have hw := (norm_nonneg z).trans_lt hz
```

Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:583]:
```suggestion
  have hw := (norm_nonneg z).trans_lt hz -- `0 < ‖w‖`
```

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 2]
```diff
@@ -562,6 +570,34 @@ theorem not_summable_of_ratio_test_tendsto_gt_one {α : Type*} [SeminormedAddCom
...
+  obtain ⟨C, hC⟩ := norm_bounded_of_cauchy_series h
+  rw [summable_iff_cauchySeq_finset]
+  refine' @cauchySeq_finset_of_geometric_bound _ _ (‖z‖ / ‖w‖) C _ _ (fun n ↦ _)
+  · rwa [div_lt_one hw]
+  · replace hC := hC n
+    rw [norm_mul, norm_pow, div_pow, ← mul_comm_div]
+    rw [norm_mul, norm_pow, ← _root_.le_div_iff (by positivity)] at hC
+    gcongr
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:591]:
```suggestion
  refine cauchySeq_finset_of_geometric_bound (r := ‖z‖ / ‖w‖) (C := C) ((div_lt_one hw).mpr hz)
    (fun n ↦ ?_)
  rw [norm_mul, norm_pow, div_pow, ← mul_comm_div]
  conv at hC => enter [n]; rw [norm_mul, norm_pow, ← _root_.le_div_iff (by positivity)]
  exact mul_le_mul_of_nonneg_right (hC n) (pow_nonneg (norm_nonneg z) n)
```
Just a suggestion...

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 3]
```diff
@@ -562,6 +570,34 @@ theorem not_summable_of_ratio_test_tendsto_gt_one {α : Type*} [SeminormedAddCom
   rwa [← le_div_iff (lt_of_le_of_ne (norm_nonneg _) h₁.symm)]
 #align not_summable_of_ratio_test_tendsto_gt_one not_summable_of_ratio_test_tendsto_gt_one
 
+section NormedDivisionRing
+
+variable [NormedDivisionRing α] [CompleteSpace α] {f : ℕ → α}
+
+/-- If a power series converges at `w`, it converges absolutely at all `z` of lesser norm. -/
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:577]:
```suggestion
/-- If a power series converges at `w`, it converges absolutely at all `z` of smaller norm. -/
```

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 4]
```diff
@@ -562,6 +570,34 @@ theorem not_summable_of_ratio_test_tendsto_gt_one {α : Type*} [SeminormedAddCom
...
+  obtain ⟨C, hC⟩ := norm_bounded_of_cauchy_series h
+  rw [summable_iff_cauchySeq_finset]
+  refine' @cauchySeq_finset_of_geometric_bound _ _ (‖z‖ / ‖w‖) C _ _ (fun n ↦ _)
+  · rwa [div_lt_one hw]
+  · replace hC := hC n
+    rw [norm_mul, norm_pow, div_pow, ← mul_comm_div]
+    rw [norm_mul, norm_pow, ← _root_.le_div_iff (by positivity)] at hC
+    gcongr
+
+/-- If a power series converges at 1, it converges absolutely at all `z` of lesser norm. -/
```
Comment (MichaelStollBayreuth) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:593]:
```suggestion
/-- If a power series converges at 1, it converges absolutely at all `z` of smaller norm. -/
```

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 5]
```diff
@@ -562,6 +569,31 @@ theorem not_summable_of_ratio_test_tendsto_gt_one {α : Type*} [SeminormedAddCom
...
 
+section NormedDivisionRing
+
+variable [NormedDivisionRing α] [CompleteSpace α] {f : ℕ → α}
+
+/-- If a power series converges at `w`, it converges absolutely at all `z` of smaller norm. -/
+theorem summable_power_of_norm_lt {w z : α}
+    (h : CauchySeq fun n ↦ ∑ i in range n, f i * w ^ i) (hz : ‖z‖ < ‖w‖) :
+    Summable fun n ↦ f n * z ^ n := by
+  have hw := (norm_nonneg z).trans_lt hz -- `0 < ‖w‖`
```
Comment (eric-wieser) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:580]: Don't put in a comment what you can put in code:
```suggestion
  have hw : 0 < ‖w‖ := (norm_nonneg z).trans_lt hz
```

**Review by MichaelStollBayreuth — APPROVED** [review]
LGTM!

File: Mathlib/Analysis/SpecificLimits/Normed.lean [hunk 6]
```diff
@@ -445,6 +445,13 @@ theorem NormedAddCommGroup.cauchy_series_of_le_geometric'' {C : ℝ} {u : ℕ 
     exact h _ H
 #align normed_add_comm_group.cauchy_series_of_le_geometric'' NormedAddCommGroup.cauchy_series_of_le_geometric''
 
+/-- The term norms of any convergent series are bounded by a constant. -/
+lemma norm_bounded_of_cauchy_series (h : CauchySeq fun n ↦ ∑ k in range n, f k) :
```
Comment (alexjbest) [inline Mathlib/Analysis/SpecificLimits/Normed.lean:449]: I think this needs to be renamed a bit, `bounded` and `cauchy_series` aren't names so this seems very hard to discover using the naming convention at present

Comment (Parcly-Taxel) [reply Mathlib/Analysis/SpecificLimits/Normed.lean:449]: I don't know; `cauchy_series` is in two other theorems in this file. `exists_norm_le_of_cauchySeq`?

Comment (alexjbest) [reply Mathlib/Analysis/SpecificLimits/Normed.lean]: I think that those should also be changed, but let's not have that hold up this pr

**Comment by alexjbest** [comment]
maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**